lib/reel/session.rb

```diff
@@ -87,5 +73,13 @@ def set_response uuid
 module Reel
   class Request
     include ::Reel::Session::RequestMixin
+
+    class Parser
+      alias_method :base_on_headers_complete, :on_headers_complete
+      def on_headers_complete headers
+        base_on_headers_complete headers
+        current_request.initialize_session
+      end
+    end
```

Comment:
this wound up being the best approach I think: rather than try to use included or extended, just open up the class and alias_method/wrap the original.